lib/imap.c

```diff
@@ -1513,10 +1522,10 @@ static CURLcode imap_done(struct Curl_easy *data, CURLcode status,
   }
   else if(!data->set.connect_only && !imap->custom &&
           (imap->uid || imap->mindex || data->state.upload ||
-          data->set.mimepost.kind != MIMEKIND_NONE)) {
+            IMAP_MIMEKIND_IS_NOT_NONE(data))) {
```

Comment:
couldn't this be a single shorter macro somewhere like urldata and make it different for when mime is disabled like 

~~~c
#ifndef CURL_DISABLE_MIME
#define IS_MIME_POST(a) ((a)->set.mimepost.kind != MIMEKIND_NONE)
#else
#define IS_MIME_POST(a) FALSE
#endif
~~~

Reply:
I've moved the macro definition to urldata.h.